File: MedicalSampleDistribution/Frontend/src/types/material.ts
export interface Material {
  id: number;
  codigoSap: string;
  description: string;
  status: string;
  currentStock?: number;
  pack?: number;
}

export interface MaterialStockInfo {
  importId: number;
  stock?: number;
  stockReal?: number;
  pack: number;
  maxStock?: number;
  minStock?: number;
  useStockReal?: boolean;
  stockManual?: number;
}

export interface MaterialDetail {
  id: number;
  codigoSap: string;
  description: string;
  status: string;
  stockByImport?: MaterialStockInfo[];
}

export interface MaterialListResponse {
  materials: Material[];
  totalItems: number;
  totalPages: number;
  currentPage: number;
}

File: MedicalSampleDistribution/Backend/MMDistributionSystem.API/Models/Material.cs
using System;
using System.Collections.Generic;

namespace MMDistributionSystem.API.Models;

public partial class Material
{
    public int Id { get; set; }

    public string CodigoSap { get; set; } = null!;

    public string Description { get; set; } = null!;

    public string Status { get; set; } = null!;
}
